docs/v1/P4Runtime-Spec.adoc

```diff
@@ -2211,73 +2034,88 @@ set, a single entry will be read.
 Let's look at the counter example in more details. Based on this specification
 document, the C++ server code which processes `CounterEntry` messages may look
 like this:
-~ Begin CPP
+
+[source,c++]
+----
 auto *counter_entry = ...
 if (counter_entry->has_index()) {
   auto index = counter_entry->index().index();
   read_one_entry(counter_entry->id(), index);
 } else {
   read_all_entries(counter_entry->id());
 }
-~ End CPP
+----
 
-1. Reading a single counter entry at index 0 in the counter array with id
+. Reading a single counter entry at index 0 in the counter array with id
 `<id>`:
     * Here is the C++ client code:
-~ Begin CPP
+
+[source,c++]
+----
 p4::v1::CounterEntry entry;
 entry.set_counter_id(<id>);
 entry.mutable_index();
 // The above line sets the index field; it is equivalent to:
 // auto *index = entry.mutable_index();
 // index->set_index(0);
-~ End CPP
+----
+
     * Here is the corresponding Protobuf message in text format:
-~ Begin Prototext
+
+[source,protobuf]
+----
 counter_id: <id>
 index {}
-~ End Prototext
+----
+
     * **Expected behavior**: Counter entry at index 0 is read. Notice that the
       `index` subfield is missing under the `index` field message of
       `CounterEntry` in the text dump of the message. This is because the
       subfield is a scalar numeric type and 0 is therefore its default
       value. Scalar fields with default values are omitted from the textual
       representation of Protobuf messages.
 
-2. Reading all counter entries by leaving the `index` field unset
+
+. Reading all counter entries by leaving the `index` field unset
```

Comment:
PDF shows a leading "1."

Reply:
if it is a 1 instead of a 2, then there must be something between the first and second item that makes asciidoctor think they are two separate numbered lists.  Perhaps the blank line between them should be removed?